nasdf/compilation-tests.lisp

```diff
@@ -37,24 +48,70 @@ A sub-package has a name that starts with that of PACKAGE followed by a '/' sepa
 (defun list-subpackages (package)
   (remove-if (lambda (pkg) (not (subpackage-p pkg package))) (list-all-packages)))
 
-(defun unbound-exports (package)
-  "Report unbound exported symbols for PACKAGE and all its subpackages."
-  ;; TODO: Only SBCL is supported for now.
-  #-sbcl
-  nil
-  #+sbcl
-  (let* ((package (find-package package))
-         (report (delete nil
-                         (mapcar (lambda (package)
-                                   (let ((exports (list-unbound-exports package)))
-                                     (when exports
-                                       (list package exports))))
-                                 (cons (find-package package) (list-subpackages package))))))
-    (when report
-      (error "~a~&Found unbound exported symbols in ~a package~:p."
-             report (length report)))))
+(defun list-undocumented-exports (package)
+  (let ((result '())
+        (classes (loop for s being the external-symbol in package
+                       when (find-class s nil)
+                         collect s)))
+    (do-external-symbols (s (find-package package) result)
+      (unless (or (some (lambda (doctype) (documentation s doctype))
+                        '(variable function compiler-macro setf method-combination type structure))
+                  ;; Check for being a slot accessor, as these often
+                  ;; have no documentation.
+                  (and (fboundp s)
+                       (typep (symbol-function s) 'generic-function)
+                       (some (lambda (class)
+                               (find-method (symbol-function s) '() (list (find-class class)) nil))
+                             classes))
```

Comment:
Should be `flet`-bound, or `defun`.

But this is not very reliable either, since any generic function with a single class specializer is matched.

c2mop can identify accessors with the `c2mop:standard-accessor-method` type.

Reply:
I really don't want to bring MOP into the picture in our dependency-less NASDF :P

Reply:
I mean we can easily copy-paste `c2mop:standard-accessor-method` into NASDF.  Here are the "MOP" packages of all  Lisps according to c2mop:

```lisp
  (:import-from
   #+abcl      #:mop
   #+allegro   #:mop
   #+clisp     #:clos
   #+clozure   #:ccl
   #+cmu       #:clos-mop
   #+ecl       #:clos
   #+clasp     #:clos
   #+lispworks #:clos
   #+mcl       #:ccl
   #+sbcl      #:sb-mop
   #+scl       #:clos
   #+mezzano   #:mezzano.clos
   #+sicl      #:sicl-clos
```

The `standard-accessor-method` type is the same symbol across all Lisps.

Reply:
Oh wow, that's impressive :D